### PR TITLE
Fix mypy errors for pandas\tests\series\test_operators.py

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
part of #28926

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

The errors were:
```
pandas/tests/series/test_operators.py:824: error: "Type[Series]" has no attribute "div"
pandas/tests/series/test_operators.py:824: error: "Type[Series]" has no attribute "rdiv"
```
The problem is that these methods are only added at RunTime, so mypy can't see them. I'm not sure if my solution is the best way to silence these errors though.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
> The problem is that these methods are only added at RunTime, so mypy can't see them. I'm not sure if my solution is the best way to silence these errors though.

they are added in add_flex_arithmetic_methods which is passed the class as a parameter. They could therefore be specific to a class and so adding them directly in Series (and adding similar in DataFrame separately) seems reasonable to me.

File: pandas/core/series.py [hunk 1]
```diff
@@ -185,6 +185,8 @@ class Series(base.IndexOpsMixin, generic.NDFrame):
         base.IndexOpsMixin.hasnans.func, doc=base.IndexOpsMixin.hasnans.__doc__
     )
     _data: SingleBlockManager
+    div: Callable[["Series"], "Series"]
+    rdiv: Callable[["Series"], "Series"]
```
Comment (simonjayhawkins) [inline pandas/core/series.py]: I assume the first argument to the Callable would be self in this case. should we have other typed?

Comment (jbrockmendel) [reply pandas/core/series.py]: other can be just about anything

Comment (SaturnFromTitan) [reply pandas/core/series.py]: @simonjayhawkins @jbrockmendel 
Does that mean the annotation should be `Callable[["Series", Any], "Series"]`?

**Review by simonjayhawkins — APPROVED** [review]
@SaturnFromTitan lgtm pending ci fixes.

**Comment by SaturnFromTitan** [comment]
@simonjayhawkins the build is green again

**Review by simonjayhawkins — APPROVED** [review]
@SaturnFromTitan lgtm @jbrockmendel @WillAyd

**Comment by WillAyd** [comment]
Thanks @SaturnFromTitan

**Pull request merged**